classes/class-pmpro-orders-list-table.php

```diff
@@ -0,0 +1,1049 @@
+                        jQuery('#end-year').hide();
+                        jQuery('#predefined-date').hide();
+                        jQuery('#status').hide();
+                        jQuery('#l').show();
+                        jQuery('#discount-code').hide();
+                        jQuery('#submit').show();
+                        jQuery('#from').hide();
+                        jQuery('#to').hide();
+                        jQuery('#filterby').show();
+                    }
+                    else if (filter == 'with-discount-code') {
+                        jQuery('#start-month').hide();
+                        jQuery('#start-day').hide();
+                        jQuery('#start-year').hide();
+                        jQuery('#end-month').hide();
+                        jQuery('#end-day').hide();
+                        jQuery('#end-year').hide();
+                        jQuery('#predefined-date').hide();
+                        jQuery('#status').hide();
+                        jQuery('#l').hide();
+                        jQuery('#discount-code').show();
+                        jQuery('#submit').show();
+                        jQuery('#from').hide();
+                        jQuery('#to').hide();
+                        jQuery('#filterby').show();
+                    }
+                    else if (filter == 'within-a-status') {
+                        jQuery('#start-month').hide();
+                        jQuery('#start-day').hide();
+                        jQuery('#start-year').hide();
+                        jQuery('#end-month').hide();
+                        jQuery('#end-day').hide();
+                        jQuery('#end-year').hide();
+                        jQuery('#predefined-date').hide();
+                        jQuery('#status').show();
+                        jQuery('#l').hide();
+                        jQuery('#discount-code').hide();
+                        jQuery('#submit').show();
+                        jQuery('#from').hide();
+                        jQuery('#to').hide();
+                        jQuery('#filterby').show();
+                    }
+                    else if(filter == 'only-paid' || filter == 'only-free' ) {
+                        jQuery('#start-month').hide();
+                        jQuery('#start-day').hide();
+                        jQuery('#start-year').hide();
+                        jQuery('#end-month').hide();
+                        jQuery('#end-day').hide();
+                        jQuery('#end-year').hide();
+                        jQuery('#predefined-date').hide();
+                        jQuery('#status').hide();
+                        jQuery('#l').hide();
+                        jQuery('#discount-code').hide();
+                        jQuery('#submit').show();
+                        jQuery('#from').hide();
+                        jQuery('#to').hide();
+                        jQuery('#filterby').hide();
+                    }
```

Comment:
```suggestion
                    }  else {
                        jQuery('#start-month').hide();
                        jQuery('#start-day').hide();
                        jQuery('#start-year').hide();
                        jQuery('#end-month').hide();
                        jQuery('#end-day').hide();
                        jQuery('#end-year').hide();
                        jQuery('#predefined-date').hide();
                        jQuery('#status').hide();
                        jQuery('#l').hide();
                        jQuery('#discount-code').hide();
                        jQuery('#submit').show();
                        jQuery('#from').hide();
                        jQuery('#to').hide();
                        jQuery('#filterby').hide();
                    }
```
If  custom filters are being  added with `pmpro_admin_orders_filters`, we want to make sure that the other filters' options don't show.